webrender/src/clip_scroll_node.rs

```diff
@@ -369,13 +369,20 @@ impl ClipScrollNode {
             resource_cache,
             device_pixel_scale,
         );
-        let (screen_inner_rect, screen_outer_rect) =
-            clip_sources.get_screen_bounds(&self.world_viewport_transform, device_pixel_scale);
+
+        let (screen_inner_rect, screen_outer_rect) = clip_sources.get_screen_bounds(
+            &self.world_viewport_transform,
+            device_pixel_scale,
+            None,
+        );
 
         // All clipping ClipScrollNodes should have outer rectangles, because they never
         // use the BorderCorner clip type and they always have at last one non-ClipOut
         // Rectangle ClipSource.
-        let screen_outer_rect = screen_outer_rect.expect("Clipping node didn't have outer rect.");
+        let screen_outer_rect = match screen_outer_rect {
```

Comment:
Do ClipScrollNodes not always have an outer rectangle now?

Reply:
In this code, it may be `None` if either `local_outer_rect` of the clip source is `None`, or the computed bounding rect is outside the screen.